src/api/app/models/user.rb

```diff
@@ -657,7 +657,7 @@ def declined_requests(search = nil)
   def incoming_requests(search = nil, states: [:new])
     result = BsRequest.where(state: states).where(id: BsRequestAction.bs_request_ids_of_involved_projects(involved_projects.pluck(:id))).or(
       BsRequest.where(id: BsRequestAction.bs_request_ids_of_involved_packages(involved_packages.pluck(:id)))
-    ).with_actions.where(state: states)
+    ).with_actions
```

Comment:
```suggestion
    result = BsRequest.where(state: states).and(
      BsRequest.where(id: BsRequestAction.bs_request_ids_of_involved_projects(involved_projects.pluck(:id))).or(
        BsRequest.where(id: BsRequestAction.bs_request_ids_of_involved_packages(involved_packages.pluck(:id)))
      )
    ).with_actions
```
Maybe this one?

Reply:
Is it only a matter of code style, or does it really change the result?

Reply:
I think your code works on the `BsRequest`, the entire set for computing the `or` condition. The original one instead first of all filters by `state` then it does the `or` part. Right?

Reply:
We can always add `.explain` at the end and compare...

Reply:
After reading and analyzing it, I like better yours, it's better defined and it makes sure the `and` and the `or` are applied properly.

I'm just splitting the query into chunks, I feel it's better readable, plus the initial filtered `state` set can be reused in the subquery. See the following:

```
result = BsRequest.where(state: states)
result = result.where(id: BsRequestAction.bs_request_ids_of_involved_projects(involved_projects.pluck(:id))).or(
  result.where(id: BsRequestAction.bs_request_ids_of_involved_packages(involved_packages.pluck(:id)))
)
result = result.with_actions
```

What do you think?

Reply:
Mmmm, forget about it!!
The first line `result = BsRequest.where(state: states)` would cost too much in multiple cases. Better to not do this separately. I'll go with your solution. :+1: